From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Giuliano Procida' via kernel-team <kernel-team@android.com>
Date: Tue, 4 Feb 2020 17:16:27 +0000
Subject: Sort kernel module object files before processing them.

This imposes a deterministic ordering, making diffs more predictable
and allowing reproducible testing.

        * src/abg-tools-utils.cc (get_binary_paths_from_kernel_dist):
        Sort module_paths.

Signed-off-by: Giuliano Procida <gprocida@google.com>
Signed-off-by: Matthias Maennich <maennich@google.com>
---
 src/abg-tools-utils.cc | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/abg-tools-utils.cc b/src/abg-tools-utils.cc
index 6608ffd5caaa..92a27d08057b 100644
--- a/src/abg-tools-utils.cc
+++ b/src/abg-tools-utils.cc
@@ -2326,7 +2326,8 @@ find_vmlinux_path(const string&	from,
 /// binary that was found.
 ///
 /// @param module_paths output parameter.  The paths of the kernel
-/// module binaries that were found.
+/// module binaries that were found, sorted to impose a deterministic
+/// ordering.
 ///
 /// @return true if at least the path to the vmlinux binary was found.
 bool
@@ -2365,6 +2366,8 @@ get_binary_paths_from_kernel_dist(const string&	dist_root,
   if (find_vmlinux_and_module_paths(from, vmlinux_path, module_paths))
     found = true;
 
+  std::sort(module_paths.begin(), module_paths.end());
+
   return found;
 }
 
